### PR TITLE
Add menu component

### PR DESCRIPTION
**Pull request by Krishna-13-cyber**
Asciinema : https://asciinema.org/a/dcgGesHQtRtQZOiCBLZCt9a81
Feature : Menu Component [Gives user menu options to user]

Signed-off-by: Krishna Narayanan <krishnanarayanan132002@gmail.com>

### CODE REVIEW

**Comment by SAtacker** [comment]
@Krishna-13-cyber please add tests

**Comment by SAtacker** [comment]
@Krishna-13-cyber Please display the asciinema in the README.md and mention the feature

**Review by SAtacker — CHANGES_REQUESTED** [review]
@Krishna-13-cyber please check clang-format workflow and resolve

File: CMakeLists.txt [hunk 1]
```diff
@@ -61,7 +61,7 @@ if(QUICK_FTXUI_TESTS)
 
     FetchContent_MakeAvailable(Catch2)
 
-    add_executable(tests tests/test1.cpp)
+    add_executable(tests tests/test1.cpp tests/test2.cpp)
```
Comment (SAtacker) [inline CMakeLists.txt]: @Krishna-13-cyber I don't think you need to create a new file

File: include/quick-ftxui.hpp [hunk 2]
```diff
@@ -31,10 +31,11 @@ struct nil {};
...
 struct expression;
 struct input;
+struct menu;
 
-typedef boost::variant<nil, boost::recursive_wrapper<button>,
-                       boost::recursive_wrapper<input>,
-                       boost::recursive_wrapper<expression>>
+typedef boost::variant<
+    nil, boost::recursive_wrapper<button>, boost::recursive_wrapper<menu>,
+    boost::recursive_wrapper<input>, boost::recursive_wrapper<expression>>
```
Comment (SAtacker) [inline include/quick-ftxui.hpp]: This change wasn't necessary. Please undo.

File: examples/Menu.qf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    Menu{
+        [ "Physics"  "Maths"  "Chemistry"  "Biology"],
+        0
+    }
+}
```
Comment (SAtacker) [inline examples/Menu.qf]: Can you please add an example with menu and button both?

**Review by SAtacker — CHANGES_REQUESTED** [review]
Please make the necessary changes.

**Comment by Krishna-13-cyber** [comment]
Closing this PR, a new one has been opened #18 with all the suggested changes

**Pull request closed without merging**